src/ds/ds-widget.js

```diff
@@ -120,13 +111,35 @@ jQuery(function ($) {
                         } else if (opts.maxResults > 0 && results.length > opts.maxResults) {
                             render(obj.options.too_many_results(this, results.length))
                         } else {
-                            for (let i in results) {
-                                let data = results[i]
-                                counter += 1;
-                                data.counter = counter;
-                                data.saved = false;
-                                render(obj.options.render_search_result(data))
+                            const getResults = function() {
```

Comment:
getResults() returns a subset of institutions from the main list of results. It simply checks what's shown in the search_result_selector and counts the number of children.